src/map/ai/states/magic_state.cpp

```diff
@@ -253,6 +253,16 @@ uint32 CMagicState::CalculateCastTime(CSpell* PSpell)
         uint16 songcasting = m_PEntity->getMod(MOD_SONG_SPELLCASTING_TIME);
         cast = cast * (1.0f - ((songcasting > 50 ? 50 : songcasting) / 100.0f));
     }
+    if (PSpell->isCure())
+    {
+        int32 cureCast = m_PEntity->getMod(MOD_CURE_CAST_TIME);
+        if (m_PEntity->objtype == TYPE_PC)
+        {
+            cureCast += ((CCharEntity*)m_PEntity)->PMeritPoints->GetMeritValue(MERIT_CURE_CAST_TIME, (CCharEntity*)m_PEntity);
+        }
+        cureCast = dsp_cap(cureCast, 0, 100);
+        cast = cast * ((100.0f - cureCast) / 100.0f);
```

Comment:
you're going to have to verify that cure cast is multiplicative with fast cast and not additive